src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -180,10 +181,26 @@ const convertFrameworkSummary = (framework) => {
 const convertTaskDetail = async (taskStatus, ports, logPathPrefix) => {
   // get container ports
   const containerPorts = {};
-  if (ports) {
+  const hashFunc = (str) => {
+    const hexStr = crypto.createHash('md5').update(str).digest('hex');
+    return parseInt(hexStr.substring(0, 12), 16) +
+      parseInt(hexStr.substring(12, 24), 16) + parseInt(hexStr.substring(24, 32), 16);
+  };
+  if (ports && taskStatus.attemptStatus.podUID) {
     const randomPorts = JSON.parse(ports);
-    for (let port of Object.keys(randomPorts)) {
-      containerPorts[port] = randomPorts[port].start + taskStatus.index * randomPorts[port].count;
+    if (randomPorts.ports) {
+      for (let port of Object.keys(randomPorts.ports)) {
+        const portNums = [...Array(randomPorts.ports[port].count).keys()].map((index) => {
+          const rawString = `[${taskStatus.attemptStatus.podUID}][${port}][${index}]`;
+          return hashFunc(rawString) % (randomPorts.portEnd - randomPorts.portStart) + randomPorts.portStart;
```

Comment:
Will hashFunc overflow then become negative? If so, make use % in js is unsigned mod operation

Reply:
It will not overflow. The max int in JS is `2^53 - 1`. For the hash func. It's `48 bit int + 48 bit int + 32 bit int` The max result is a `49 bit int`

Reply:
It will not overflow. The max int in JS is `2^53 - 1`. For the hash func. It's `48 bit int + 48 bit int + 32 bit int` The max result is a `49 bit int`